jax/numpy/lax_numpy.py

```diff
@@ -1243,6 +1243,57 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
   else:
     return (ar1[:, None] == ar2).any(-1)
 
+@partial(jit, static_argnums=2)
+def _intersect1d_sorted_mask(ar1, ar2, return_indices=False):
+    """
+    Helper function for intersect1d which is jit-able
+    """
+    ar = concatenate((ar1, ar2))
+
+    if return_indices:
+      indices = argsort(ar)
+      aux = ar[indices]
```

Comment:
It's possible to use `lax.sort` directly to get the sorted array and the indices in a single pass (if I recall correctly, argsort uses this and then discards the sorted array).